### DISCUSSION TITLE
Constant CPU Use

### DISCUSSION THREAD

**Discussion by bdkjones** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#818] [General]
**Describe the bug**

Al Dente constantly uses about 0.5-0.8 CPU, which is wasteful.

**To Reproduce**
1. Run Al Dente.

**Expected behavior**
Al Dente should not consume CPU cycles needlessly.

**Screenshots**
<img width="1054" alt="Screen Shot 2022-09-03 at 16 10 55" src="https://user-images.githubusercontent.com/926566/188290264-ee09d6dc-8a80-47af-a0d4-3c32aa861fef.png">

**MacBook:**
 - MacBook Pro (16-inch, 2021)
 - macOS 12.5.1
 - AlDente Free
 - AlDente 1.18
 - Default settings enabled; charge level set to 69% and maintaining.

**Additional context**
This looks like the sort of wasteful CPU usage you get with polling. I haven't looked through the codebase extensively, but I'd wager there's polling going on.  A constant use of 1% CPU for an app that simply maintains a certain battery charge level seems wasteful.

I'm also curious what the `AlDente Networking` process is and why there's a constant process dedicated to the apphousekitchen website. I understand checking for updates, but those processes should be closed and cleaned up once the update check completes. I don't understand why these two processes must exist at all times.

In short, the app seems to be a bit careless with system resources.

**Comment by MatthiasKerbl** [comment]
Hi @bdkjones,

For me, the CPU use is between 0.3-0.5%. Unfortunately, polling is necessary since AlDente has to actively monitor the MacBook to alter the charging behavior. However, compared to other little macOS apps such as TinyStopwatch or CheatSheet, which use roughly 0.8% of CPU on my system, AlDentes CPU usage is within a reasonable range.

Regarding the Network processes, I do not know where exactly they are coming from. They do not appear on our test systems. It might have something to do with the licensing server or the update feature.

Hope that helps.

Cheers,
Matthias

**Comment by bdkjones** [comment]
Great; thanks for the info!

Is it possible to increase the polling interval? It would be nice to save all the CPU cycles we can.

**Comment by MatthiasKerbl** [comment]
> 

Hi @bdkjones,

It would theoretically be possible to increase the polling interval. However, AlDente would become less responsive. Therefore, we had to find a sweet spot between responsiveness and CPU usage.

Cheers,
Matthias

**Reply by mailinglists35** [reply]
> It would theoretically be possible to increase the polling interval. However, AlDente would become less responsive. Therefore, we had to find a sweet spot between responsiveness and CPU usage.

can you offer user tuning of polling interval? I really do not need that continuous monitoring. One sampling per minute would be sufficient, and let the app really do nothing the other 59 seconds please.

It is a waste of cpu and energy and electricity bill (if you multiplty to number of aldente instances worldwide).

**Comment by MatthiasKerbl** [comment]
Hi @bdkjones,

I am converting this issue into a discussion since it is more of a general discussion than an actual issue.

Hope that's understandable.

Cheers,
Matthias

**Comment by jordanade** [comment]
On my M2 Air, it's constantly using between 17% and 25% CPU! How can I resolve this?
<img width="642" alt="Screenshot 2024-04-23 at 7 16 53 PM" src="https://github.com/AppHouseKitchen/AlDente-Charge-Limiter/assets/2799445/29dca36a-f3f3-4ef6-9ffd-2b78c72bd4f5">

**Reply by jordanade** [reply]
Update: I opened and closed the config window and it went down to under 1%.

**Comment by karroze** [comment]
Same behavior as the commenter above: huge CPU usage, open config window, close it (via red button) and it does way below 1%

<img width="547" alt="Screenshot 2025-03-22 at 23 24 35" src="https://github.com/user-attachments/assets/fa00d030-91a5-49e1-96f8-3ede3f885409" />

**Reply by MatthiasKerbl** [reply]
Hi @karroze,

Thank you for reaching out!

What you’re experiencing is completely normal. The temporary increase in CPU usage happens due to the various diagrams and graphs displayed in the dashboard. As you’ve already noticed, once you close the window, AlDente’s CPU usage drops to a minimal level.

Best,
Matthias

**Reply by karroze** [reply]
Hi! Unfortunately, this is not the case I'm trying to describe. 

Here are detailed steps to reproduce:

1. Launch AlDente with "Show dock icon" option disabled
2. Open dashboard from system tray icon
3. CMD+Tab, which will close AlDente window 
4. Wait for some time, after which AlDente will take 20-30-40% of CPU
5. Open dashboard from system tray, close it with red button
6. Issue is gone, AlDente is back to <1% of CPU usage

So it seems that CMD+Tab closing of the app's window causes this problem some time after 

Hope this info helps 😄

**Reply by karroze** [reply]
Since my last comment I've been closing the app by pressing "red button", instead of `CMD+Tab`-ing out of it, and the issue is gone. So there's definitely something up with that 😄
